Userland/Applications/HexEditor/HexEditorWidget.cpp

```diff
@@ -330,7 +333,28 @@ void HexEditorWidget::update_inspector_values(size_t position)
         value_inspector_model->set_parsed_value(ValueInspectorModel::ValueType::Double, "");
     }
 
-    // FIXME: Parse as other values like ASCII, UTF8, UTF16, Timestamp etc
+    // FIXME: This probably doesn't honour endianness correctly.
```

Comment:
This should probably get _some_ attention in this PR already - are you showing UTF-16BE or UTF-16LE? It's probably arch-dependent because of the `u16 const*` cast. Maybe you can show both BE and LE by manually reconstructing the `u16` values from an `u8 const*`?

Reply:
I do think that I'll be considering UTF-16 endianness correctly, as the numbers are converted to host endianness correctly before we read them in here. However, because we're reading 8 bytes at a time, the indicated UTF-16 code unit is several bytes after the cursor for big endian (IIRC, maybe the other way around)

Reply:
Ah that `unsigned_64_bit_int` is already taking into account the value of `m_value_inspector_little_endian`. Maybe, for clarity's sake, it would be best to format the caption for the UTF-16 value as "UTF-16LE" or "UTF-16BE" depending on this setting?

Reply:
I think we should have better indication of endianness in the value inspector in general, which would also make it clearer for all other values.

Reply:
Marking this as resolved, there's an issue for it now: #13653